### PR TITLE
docs: add doc for auto rotation

### PR DESCRIPTION
**Pull request by aramase**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, using `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when the PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

/kind documentation

### CODE REVIEW

File: docs/README.rotation.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Auto rotation of mounted contents and synced Kubernetes Secrets
+
+- **Design doc:** [Rotation Design](https://docs.google.com/document/d/1RGT0vmeUnN71n_u5fZKsSCa2YQpGw99rfGN9RlFMgHs/edit?usp=sharing)
+- **Feature State:** Secrets Store CSI Driver v0.0.15 [alpha] 
+
+When the secret/key is updated in external secrets store after the inital pod deployment, the updated secret will be periodically updated in the pod mount and the Kubernetes Secret.
+
+Depending on how the application consumes the secret data:
+1. **Mount Kubernetes secret as a volume:** Use auto rotation feature + Sync K8s secrets feature in Secrets Store CSI Driver, application will need to watch for changes from the mounted Kubernetes Secret volume. When the Kubernetes Secret is updated by the CSI Driver, the corresponding volume conteents are automatically updated.
```
Comment (ritazh) [inline docs/README.rotation.md:9]: s/conteents/contents

File: docs/README.rotation.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Auto rotation of mounted contents and synced Kubernetes Secrets
+
+- **Design doc:** [Rotation Design](https://docs.google.com/document/d/1RGT0vmeUnN71n_u5fZKsSCa2YQpGw99rfGN9RlFMgHs/edit?usp=sharing)
+- **Feature State:** Secrets Store CSI Driver v0.0.15 [alpha] 
+
+When the secret/key is updated in external secrets store after the inital pod deployment, the updated secret will be periodically updated in the pod mount and the Kubernetes Secret.
```
Comment (ritazh) [inline docs/README.rotation.md:6]:
```suggestion
When the secret/key is updated in external secrets store after the initial pod deployment, the updated secret will be periodically updated in the pod mount and the Kubernetes Secret.
```

File: docs/README.rotation.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Enable auto rotation
+
+> NOTE: This alpha feature is not enabled by default.
+
+To enable auto rotation, enable the `--enable-secret-rotation` feature gate for the `secrets-store` container in the Secrets Store CSI Driver pods. The rotation poll interval can be configured using `--rotation-poll-interval`. The default rotation poll interval is `2m`. If using helm to install the driver, set `enableSecretRotation: true` and configure the rotation poll interval by setting `rotationPollInterval`. The rotation poll interval can be tuned based on how frequently the mounted contents for all pods and Kubernetes secrets need to be resynced to the latest.
+
+- The Secrets Store CSI Driver will update the pod mount and the Kubernetes Secret defined in `secretObjects` of SecretProviderClass periodically based on the rotation poll interval to the latest value.
+- If the `SecretProviderClass` is updated after the pod was initially created, then the pod mount and Kubernetes secret will be updated with the new objects added to the `SecretProviderClass`.
```
Comment (ritazh) [inline docs/README.rotation.md:21]: Can you pls clarify if the reconciler is going to update new secrets store objects or new `secretObjects` or both?

File: docs/README.rotation.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+The CSI driver **does not restart** the application pods. It only handles updating the pod mount and Kubernetes secret similar to how Kubernetes handles updates to Kubernetes secret mounted as volumes.
```
Comment (ritazh) [inline docs/README.rotation.md:23]: We should add this as a NOTE section to the main readme under the `Enable Auto Rotation of Secrets` section

File: docs/README.rotation.md (hunk 3 above)
Comment (aramase) [reply docs/README.rotation.md]: Added more text to clarify this!

File: docs/README.rotation.md (hunk 4 above)
Comment (aramase) [reply docs/README.rotation.md:25]: Good suggestion. I've added this as a note in the main readme.

**Pull request merged**